Source/JavaScriptCore/dfg/DFGOperations.cpp

```diff
@@ -2854,6 +2866,10 @@ JSC_DEFINE_JIT_OPERATION(operationStringReplaceStringStringWithTable8, JSString*
     JITOperationPrologueCallFrameTracer tracer(vm, callFrame);
     auto scope = DECLARE_THROW_SCOPE(vm);
 
+    if (JSString* result = tryReplaceOneCharUsingString(globalObject, stringCell, searchCell, replacementCell))
```

Comment:
When using Table, we already ensured that search string is longer than 1, so let's remove this code from this path.